surfactant/database_manager/utils.py

```diff
@@ -0,0 +1,208 @@
+        tomlkit.dump(data, f)
+
+
+def load_db_version_metadata(
+    version_file_path: Union[str, Path], database_key: str
+) -> Optional[Dict[str, str]]:
+    """
+    Load the database version metadata for a specific database from the specified TOML file.
+
+    Args:
+        version_file_path (Union[str, Path]): The path to the TOML file that tracks database versions.
+        database_key (str): The key identifying the database.
+
+    Returns:
+        Optional[Dict[str, str]]: A dictionary with metadata for the database, or None if not found.
+
+        Example structure:
+        {
+            "retirejs": {
+                "file": "js_library_patterns_retirejs.json",
+                "source": "https://example.com/source.json",
+                "hash": "abc123...",
+                "timestamp": "2025-02-10T19:18:34.784116Z"
+            },
+            "abc": {
+                "file": "some_other_library_patterns_abc.json",
+                "source": "https://example.com/other_source.json",
+                "hash": "def456...",
+                "timestamp": "2025-02-10T20:00:00.000000Z"
+            }
+        }
+    """
+    db_metadata = _read_toml_file(version_file_path)
+    if db_metadata is None:
+        return None
+
+    return db_metadata.get(database_key, {})
+
+
+def save_db_version_metadata(version_info: Union[str, Path], database_info: Dict[str, str]) -> None:
+    """
+    Save the metadata for a specific database to the specified TOML file.
+
+    Args:
+        version_info (Union[str, Path]): The path to the TOML file.
+        database_info (Dict[str, str]): A dictionary containing keys:
+            - "database_key": The key identifying the database  (e.g., "retirejs")..
+            - "database_file": The file name of the database (e.g., "js_library_patterns_retirejs.json").
+            - "source": The source URL of the database.
+            - "hash_value": The hash value of the database file.
+            - "timestamp": The timestamp when the database was downloaded.
+
+    Raises:
+        ValueError: If required keys are missing from database_info.
+    """
+    required_keys = {"database_key", "database_file", "source", "hash_value", "timestamp"}
+    if not required_keys.issubset(database_info):
+        raise ValueError(f"database_info must contain the keys: {required_keys}")
+
+    db_metadata = _read_toml_file(version_info) or {}
+    new_data = {
+        database_info["database_key"]: {
+            "file": database_info["database_file"],
+            "source": database_info["source"],
+            "hash": database_info["hash_value"],
+            "timestamp": database_info["timestamp"],
+        }
+    }
+    db_metadata.update(new_data)
+    _write_toml_file(version_info, db_metadata)
+
+
+# URL for the hosted external TOML file on ReadTheDocs
+DEFAULT_EXTERNAL_DB_CONFIG_URL = (
+    "https://readthedocs.org/projects/surfacet-docs/downloads/latest/database_sources.toml"
```

Comment:
This URL doesn't look like it will be valid. Should be something under https://surfactant.readthedocs.io/en/latest/